docs/community/devstack.md

```diff
@@ -0,0 +1,139 @@
+
+#### docker-compose
+Inside .compose.env, you'll find a commented out line referencing ANSIBLE_HUB_UI_PATH. If you want to add a UI to the compose stack, this needs to be uncommented and set to a valid absolute path for a checkout of [ansible-hub-ui](https://github.com/ansible/ansible-hub-ui). The compose spin up should properly allocate an alpine container with the appropriate node.js version and install+build+launch the UI.
+
+#### starting directly
+
+For development work on the UI, it's easier to launch the webpack dev server outside the compose stack. The npm start scripts and configuration will have it redirect api calls to http://localhost:5001 by default, so no hacking is necessary to make it work locally.
+
+1. Install [NVM](https://github.com/nvm-sh/nvm)
+2. Use NVM to install and set the version to 16. "nvm install 16"
+3. Clone the [ansible-hub-ui](https://github.com/ansible/ansible-hub-ui) repo to the desired location.
+4. From the checkout, run "npm install"
+5. From the checkout, run "npm run start-community"
+
+## Testing
+
+The GalaxyNG project is a "test driven development" (TDD) project. Our interpretation of TDD is that -every- PR to the master branch -must- have some sort of test or test changes. It can be unit, functional or integration, but something must be written. It doesn't matter if the PR is a bugfix or a feature, tests must be written. This is how we avoid regressions and keep track of how new code is meant to work inside a very complex and spread out architecture with many different configurable behaviors.
+
+### Unit
+
+More focus has been spent on the integration tests, but there are a few unit tests in the project. The compose stack must be running to launch unit tests, as they are using django and postgres with a temporary database to do real SQL calls that are incompatible with sqlite.
+
+To launch units, execute "make docker/test/unit". Specific tests can be selected by running "TEST=xyz make docker/test/unit".
+
+
+### Functional
+
+Pulp has a testing concept known as [functional tests](https://docs.pulpproject.org/pulpcore/en/master/nightly/contributing/tests.html#functional-tests). These are mostly using a suite of fixtures, auto-generated api clients and strict style to do integration testing. The environment needed to run functional tests is somewhat complicated to setup, so the GalaxyNG project instead writes integration tests. However, if you end up writing patches for any of the repositories in https://github.com/pulp, you will be required to write functional tests.
```

Comment:
Can get rid of "mostly" in "These are mostly using a suite of fixtures, auto-generated api clients and strict style to do integration testing."

Need extra comma there?
"These are using a suite of fixtures, auto-generated api clients,  and a strict style to do integration testing."

Reply:
ditto can likely ditch the "somewhat" in "The environment needed to run functional tests is somewhat complicated to setup"

(heh, there is a little irony in me pointing out words like "somewhat" or "mostly" given that I do that constantly in everything I write :smile: )